### PR TITLE
OHRM-72, OHRM-73: CLI bug fixes

### PR DESCRIPTION
**Pull request by MadhukaHarith92**

### CODE REVIEW

File: installer/DetailsHandler.php [hunk 1]
```diff
@@ -126,28 +126,34 @@ public function getAdminEmployeeLastName()
      * Get the organization email address
      * @return mixed
      */
-    public function getOrganizationEmailAddress()
+    public function getOrganizationEmailAddress($invalid = false)
```
Comment (samanthajayasinghe) [inline installer/DetailsHandler.php]: Don't pass the parameter to the function and this is a anti-pattern.
collect all the errors to a private variable in a class

File: installer/DetailsHandler.php [hunk 2]
```diff
@@ -126,28 +126,34 @@ public function getAdminEmployeeLastName()
...
     }
 
     /**
      * Get the Admin employee contact number
-     * @return mixed
+     * @param bool $invalid
+     * @return mixed|string
      */
-    public function getContactNumber()
+    public function getContactNumber($invalid = false)
```
Comment (samanthajayasinghe) [inline installer/DetailsHandler.php]: same here

File: installer/DetailsHandler.php [hunk 3]
```diff
@@ -126,28 +126,34 @@ public function getAdminEmployeeLastName()
...
     {
         $configurationDataSet = $this->getCongigIni();
+        $contactNumber = $_SESSION['defUser']['contactNumber'] ? $_SESSION['defUser']['contactNumber'] : $configurationDataSet["contactNumber"];
 
-        if ($_SESSION['defUser']['contactNumber']) {
-            return $_SESSION['defUser']['contactNumber'];
+        if ($invalid) {
```
Comment (samanthajayasinghe) [inline installer/DetailsHandler.php]: use something like empty($this->error['contact-number'])

File: installer/DetailsHandler.php [hunk 4]
```diff
@@ -327,6 +333,59 @@ function isFillInConfig($SessionStatus, $inputType)
...
+        return $returnValue;
+    }
+
+    /**
+     * Validate input
+     * @param $inputType
+     * @param $inputValue
+     * @return bool|mixed
+     */
+    function isValidUserInput($inputType, $inputValue) {
```
Comment (samanthajayasinghe) [inline installer/DetailsHandler.php]: Should be private function

File: installer/DetailsHandler.php [hunk 5]
```diff
@@ -327,6 +333,59 @@ function isFillInConfig($SessionStatus, $inputType)
...
+            return filter_var($inputValue, FILTER_VALIDATE_EMAIL);
+        }
+
+        if ($inputType == 'contact number') {
+            if (!($inputValue == '')) {
+                if(preg_match("/^\+?[0-9]+$/", $inputValue)) {
+                    $_SESSION['defUser']['contactNumber'] = $inputValue;
+                    return true;
+                }
+                return false;
```
Comment (samanthajayasinghe) [inline installer/DetailsHandler.php]: populate error messages and don't return it .. $this->error['contact-number'] = true

File: installer/DetailsHandler.php (hunk 1 above)
Comment (MadhukaHarith92) [reply installer/DetailsHandler.php]: updated

File: installer/DetailsHandler.php (hunk 2 above)
Comment (MadhukaHarith92) [reply installer/DetailsHandler.php]: updated

File: installer/DetailsHandler.php (hunk 3 above)
Comment (MadhukaHarith92) [reply installer/DetailsHandler.php]: updated

File: installer/DetailsHandler.php (hunk 4 above)
Comment (MadhukaHarith92) [reply installer/DetailsHandler.php]: updated

File: installer/DetailsHandler.php (hunk 5 above)
Comment (MadhukaHarith92) [reply installer/DetailsHandler.php]: updated

**Pull request merged**